announcement/core/views.py

```diff
@@ -124,166 +173,239 @@ def __init__(self, ctx: commands.Context, announcement: AnnouncementModel, *, ti
+            self.mentionable_select = MentionableSelect(
+                min_values=0,
+                max_values=25,
+                placeholder="Other mentions",
+                disabled=True,
+                row=1,
+            )
+            self.add_item(self.mentionable_select)
+            description = "__**Select mentions:**__\n"
+        elif current == "channel":
+            post = True
+            channel_select = ChannelSelect(
+                channel_types=[discord.ChannelType.news, discord.ChannelType.text],
+                placeholder="Select a channel",
+                row=0,
+            )
+            self.add_item(channel_select)
+            description = "__**Select a channel:**__\n"
+        else:
+            raise ValueError(f"Invalid session in `_action_next`: `{current}`.")
+        description += f"{self.session_description}\n"
+        embed = self.message.embeds[0]
+        embed.description = description
+        self.generate_buttons(post=post)
+        await self.update_view()
+
+    async def _action_edit(self, *args: Tuple[Interaction, Button]) -> None:
+        interaction, _ = args
+        modal = Modal(self, self.inputs)
         await interaction.response.send_modal(modal)
         await modal.wait()
 
-    async def _action_preview(self, interaction: Interaction) -> None:
+    async def _action_preview(self, *args: Tuple[Interaction, Button]) -> None:
+        interaction, _ = args
         try:
             await interaction.response.send_message(ephemeral=True, **self.announcement.send_params())
         except discord.HTTPException as exc:
             error = f"**Error:**\n```py\n{type(exc).__name__}: {str(exc)}\n```"
             await interaction.response.send_message(error, ephemeral=True)
 
-    async def _action_cancel(self, interaction: Interaction) -> None:
-        self.announcement.posted = False
+    async def _action_cancel(self, *args: Tuple[Interaction, Button]) -> None:
+        interaction, _ = args
+        self.announcement.cancel()
         self.disable_and_stop()
         await interaction.response.edit_message(view=self)
 
-    async def _action_yes(self, interaction: Interaction) -> None:
+    async def _action_confirmation(self, interaction: Interaction, button: Button) -> None:
         await interaction.response.defer()
-        self.confirm = True
-        self.disable_and_stop()
-
-    async def _action_no(self, interaction: Interaction) -> None:
-        await interaction.response.defer()
-        self.confirm = False
+        value = button.label.lower()
+        self.confirm = value == "yes"
         self.disable_and_stop()
 
     async def interaction_check(self, interaction: Interaction) -> bool:
         if self.user.id == interaction.user.id:
             return True
-        await interaction.response.send_message(
+
+        await interaction.followup.send(
             "This panel cannot be controlled by you!",
             ephemeral=True,
         )
```

Comment:
This won't work. Interaction must be responded or deferred before followup can be sent.
But here I think it would be better if we just do nothing.


```suggestion

```